cpp/unreal_plugins/UrdfBot/Source/UrdfBot/UrdfJointComponent.cpp

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "UrdfJointComponent.h"
+
+#include <iostream>
+
+#include <PhysicsEngine/PhysicsConstraintComponent.h>
+
+#include "CoreUtils/Assert.h"
+#include "UrdfLinkComponent.h"
+#include "UrdfParser.h"
+
+void UUrdfJointComponent::initialize(UrdfJointDesc* joint_desc, UUrdfLinkComponent* parent_link, UUrdfLinkComponent* child_link)
+{
+    this->joint_type_ = joint_desc->type_;
+
+    ConstraintInstance.SetDisableCollision(true);
+
+    ConstraintInstance.ProfileInstance.ConeLimit.Swing1Motion = EAngularConstraintMotion::ACM_Locked;
+    ConstraintInstance.ProfileInstance.ConeLimit.Swing2Motion = EAngularConstraintMotion::ACM_Locked;
+    ConstraintInstance.ProfileInstance.TwistLimit.TwistMotion = EAngularConstraintMotion::ACM_Locked;
+    ConstraintInstance.UpdateAngularLimit();
+
+    ConstraintInstance.ProfileInstance.AngularDrive.AngularDriveMode = EAngularDriveMode::TwistAndSwing;
+    ConstraintInstance.ProfileInstance.ConeLimit.bSoftConstraint = true;
+    ConstraintInstance.ProfileInstance.TwistLimit.bSoftConstraint = false;
+
+    const float M_TO_CM = 100.0f;
+
+    float range = (joint_desc->upper_ - joint_desc->lower_) * 0.5f;
+    float spring = joint_desc->damping_ * M_TO_CM * 10.0f;
+    float damping = joint_desc->damping_ * M_TO_CM;
+    float force_limit = joint_desc->effort_ * M_TO_CM;
```

Comment:
I dont find a corresponding value for `spring` in urdf joint, and `damping * 10` gives a reasonably good behavior to start with.

Reply:
Put this in a comment.

Reply:
fixed